lib/node_modules/@stdlib/lapack/base/dlassq/lib/base.js

```diff
@@ -0,0 +1,165 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var pow = require( '@stdlib/math/base/special/pow' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var sqrt = require( '@stdlib/math/base/special/sqrt' );
+
+
+// VARIABLES //
+
+var SBIG = 1.11137937474253874e-162;
+var SSML = 4.49891379454319638e+161;
+var TBIG = 1.99791907220223503e+146;
+var TSML = 1.49166814624004135e-154;
+
+
+// MAIN //
+
+/**
+* Returns the values `scale_out` and `sumsq_out` such that `(scale_out**2)*sumsq_out = x( 1 )**2 +...+ x( n )**2 + (scale**2)*sumsq` where `x( i ) = X( 1 + ( i - 1 )*INCX )`.
+*
+* @private
+* @param {NonNegativeInteger} N - number of elements to be used from `x`
+* @param {Float64Array} X - input array
+* @param {integer} strideX - stride length for `X`
+* @param {NonNegativeInteger} offsetX - starting index for `X`
+* @param {number} scale - scaling factor for the sum of squares
+* @param {number} sumsq - basic sum of squares from which output is factored out
+* @param {Float64Array} out - output array containing scale `scale_out` and `sumsq_out`
+* @param {integer} strideOut - stride length for `out`
+* @param {NonNegativeInteger} offsetOut - starting index for `out`
+* @returns {Float64Array} out
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var out = new Float64Array( [ 0.0, 0.0 ] );
+* var X = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+*
+* dlassq( 4, X, 1, 0, 1, 0, out, 1, 0 );
+* // out => <Float64Array>[ 1.0, 30.0 ]
+*/
+function dlassq( N, X, strideX, offsetX, scale, sumsq, out, strideOut, offsetOut ) { // eslint-disable-line max-len
```

Comment:
```suggestion
function dlassq( N, X, strideX, offsetX, out, strideOut, offsetOut ) { // eslint-disable-line max-len
```
Instead of explicitly providing `scale` and `sumsq`, do you wish to pass it via the same array as input and update it accordingly.

cc @kgryte

Reply:
No, they should be passed separately as scalar values. The output array should be considered distinct. If someone wants to overwrite, they can do

```javascript
dlassq( N, X, strideX, offsetX, out[0], out[1], out, strideOut, offsetOut )
```